### ISSUE TITLE
Api prevents access to GET requests without Content-Type

### ISSUE THREAD

**Issue by Thalhammer** [CTFd/CTFd#1974]
**Environment:** CTFD in docker using official container
**CTFd Version/Commit:** ctfd/ctfd:latest
**Operating System:** docker
**Web Browser and Version:** N/A
**What happened?**
HTML response with login page

**What did you expect to happen?**
JSON result for the api request (`/api/v1/challenges` in my case, but really any get)

**How to reproduce your issue**
Try to get a json api on any ctfd instance using "Authorizaion: Token <token>" and without session cookies.

Today I tried to install a challenge using ctfcli and got greeted by an error message. I have it attached below, but it is not really relevant as the issue seems to be with ctfd and not the api. I tried the same request using a http client and got the login page instead even though the token clearly was valid in the right format. So I dug into the code and found this line, which only accepts tokens if the content-type is set to application/json, which for a GET request is not the case. I tried it and indeed adding the "Content-Type" header to my request made it work.

- If this is expected behaviour (which I highly doubt) then this is a documentation issue since it is not documented anywhere.
- If it is not a documentation issue (more likely since my rest tool even warned me that content-type is not supposed to be on a get request), then [this](https://github.com/CTFd/CTFd/blob/021e151946fabe2d854ad91211b433739a433c1b/CTFd/utils/initialization/__init__.py#L264) is the relevant line for the check.

Note: This is hard to test in a browser since any successfully request sets the session cookie which takes precedence over the Auth header.

[traceback.txt](https://github.com/CTFd/CTFd/files/6960916/traceback.txt)

**Comment by ColdHeat** [comment]
ctfcli should already be setting the appropriate headers to communicate with CTFd. if it isn't communicating properly then the issue is with ctfcli. 

Perhaps `Content-Type` is incorrect but I don't think it should be changed, perhaps just an additional check for `Accept`. I don't think the HTTP spec enforces that GET requests can't have `Content-Type` either.

**Comment by Thalhammer** [comment]
You are right I could not find a RFC that says "you can't use content-type without a body". That said, I have also never seen any api that requires having a content-type set without having a body. Usually it's as follows:
- Set content-type if there is a body in the message (post,put,patch,...)
- Don't care if there is none (get & usually delete)
- Set Accept to indicate the content type you want back

If ctfd chooses to always require a content-type (which it is obviously free to do, albeit unusual) it should be clearly documented in the api docs. Because right now there is no mention of this at all, unless you dig deep into the code and all indicators ctfd gives is that the token is invalid or the authorization header is wrong.

**Comment by ColdHeat** [comment]
While thinking about this I also wanted to note that part of the reason this was implemented this way was to act as a security defense against potential CSRF issues. Although it has failed in the past before, generally it is not possible for a CSRF attack to forge a request containing a custom Content-Type.

**Comment by Thalhammer** [comment]
> While thinking about this I also wanted to note that part of the reason this was implemented this way was to act as a security defense against potential CSRF issues.

I don't csrf is an issue for API requests though since you would need the Token. For everything other than API requests the header is irrelevant, since it is bypassed once a valid login cookie is sent (which would be the case in a csrf).

**Comment by ColdHeat** [comment]
It is a concern for CTFd b/c the API is used to service CTFd itself as well as the API requests. It's not an issue anymore as there is now a CSRF header, but it's worth noting here I think.

**Comment by Thalhammer** [comment]
> It is a concern for CTFd b/c the API is used to service CTFd itself as well as the API requests

I don't think the check would have helped in this case anyway, because once you have a login cookie set (as you typically would have if you ever logged in) the check does not apply anymore. I noticed this cause as soon as I did one correct request with the header set I got the cookie and now the request without the header worked as well, which drove me almost insane (my client does not make auto set cookies obvious). So any csrf that would have worked without the header would work anyway if the user ever logged in (and if he did not, e.g. its public data I don't see the issue).

**Comment by nikitastupin** [comment]
Hey 👋 Just made a PR to clarify this behaviour in the documentation.
